tap_google_ads/streams.py

```diff
@@ -141,12 +150,15 @@ def on_giveup_func(err):
 
 @backoff.on_exception(backoff.expo,
                       (GoogleAdsException,
+                       InternalServerError, MethodNotImplemented,
```

Comment:
If  ServerError is the parent class of InternalServerError, MethodNotImplemented, BadGateway, ServiceUnavailable, GatewayTimeout, can pass ServerError directly here instead of too many classes?

Reply:
ServerError is the parent class of some other errors also like DataLoss, Unknown, etc ([Reference](https://github.com/googleapis/python-api-core/blob/main/google/api_core/exceptions.py)) and we do not want to retry those errors. Thats' why we can not use ServerError directly.

Reply:
Updated error handling to retry all `ServerError`.